### PR TITLE
Update docs for styles (Enhance Styles 5.0)

### PR DESCRIPTION
**Pull request by colepeters**
- Break out new substructure for /styling
- Update utilities doc
- Update element styles doc
- Add docs for modular scale, logical properties
- Update `code` styles in doc-callout

### CODE REVIEW

File: app/docs/md/learn/concepts/styling/customization.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+```arc
+@enhance-styles
+config styleguide.json
+```
+
+<doc-link-callout link="https://github.com/enhance-dev/enhance-styles#readme" mark="💅🏽">
+  Read more about the available styleguide customizations
+</doc-link-callout>
+
```
Comment (colepeters) [inline app/docs/md/learn/concepts/styling/customization.md:28]: This doc untouched, just moved to its own file.

File: app/docs/md/learn/concepts/styling/modular-scales.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+When reviewing or customizing the `typeScale` and `spaceScale`, the key concepts to be aware of are:
+
+### Steps
+
+Type and space scales contain a configurable number of steps. This number should be large enough to provide the designer and developer with a sufficient range of options for setting type size and layout spacing, but not so large that an excessive number of unused steps are generated (as this will bloat the CSS bundle and cause confusion for implementers).
+
+<doc-callout level="info">
+
+For **type scales**, the `steps` option indicates the base step plus the number of positive steps to be generated. For example, 6 steps would produce 1 base step plus 5 positive steps. **Two negative steps will also be generated**, for setting type at sub-body copy sizes.
+
```
Comment (kristoferjoseph) [inline app/docs/md/learn/concepts/styling/modular-scales.md:42]: Please call out **why** only two negative steps are generated. As the type becomes illegible at too small a size.

File: app/docs/md/learn/concepts/styling/modular-scales.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+Type and space scale values will only be fluidly interpolated between a declared minimum and maximum viewport width. Beyond these boundary sizes, the scale values will remain at their respective minimum and maximum sizes.
+
+### Base sizes
+
+The configuration’s base sizes determine the starting value to use for the scale. Each step on the scale will get larger than this size (or smaller, for negative steps) by an amount dictated by the current viewport width and the minimum and maximum scale factors.
+
+<doc-callout level="tip">
+
+Enhance Styles will automatically assign the base step of the type scale (`--text0`) to the `body` element’s `font-size` property. Setting this property to another value will mean that your default font size will not adhere to the type scale. To change the body font size, change your `typeScale.baseMin` and `typeScale.baseMax` properties accordingly.
```
Comment (kristoferjoseph) [inline app/docs/md/learn/concepts/styling/modular-scales.md:61]: This is good. Great explanation.

**Review by kristoferjoseph — COMMENTED** [review]
This is great. Only one clarification needed. The "why" only two negative steps in the type scale for users completely new to scales.

**Review by kristoferjoseph — APPROVED** [review]

**Pull request merged**